### PR TITLE
[TextFields] Add setHelperText:helperAccessibilityLabel

### LINKED ISSUE #2841: [TextFields] `helperText` should have an accessibility analogue
### Overview
The `helperText` property on `MDCTextInputController` has no accessibility equivalent. This makes it difficult for clients to set different values for the displayed value and what VoiceOver reads.

### Expected behavior

As an example, formatting a number may be handled differently for displayed and VoiceOver values:

**Displayed text**
"+82-2-312-3456 or 02-312-3456"

**VoiceOver Label**
"The domestic or international format for the telephone number."

### Reproduction steps
 - Detailed steps to reproduce

### Version number
44.3.0

### PR DESCRIPTION
**Pull request by andrewoverton**
Closes #2841.

### CODE REVIEW

File: components/TextFields/src/MDCTextInputController.h [hunk 1]
```diff
@@ -298,4 +298,22 @@
...
+ If the TextField is in an error state helperText is saved as the previousLeadingText, and
+ helperAccessibilityLabel is saved in an instance variable. When the TextField eventually leaves the
+ error state the previousLeadingText becomes the leadingUnderlineLabel's text and the
+ helperAccessibilityLabel becomes the leadingUnderlinLabel's accessibilityLabel.
+
+ If the TextField is not in an error state helperText is set as the leadingUnderlineLabel's text and
+ helperAccessibilityLabel is set as the leadingUnderlineLabel's accessibilityLabel.
+ */
+-(void)setHelperText:(nullable NSString *)helperText
+    helperAccessibilityLabel:(nullable NSString *)helperAccessibilityLabel;
```
Comment (willlarche) [inline components/TextFields/src/MDCTextInputController.h:317]: Why is this one called label and the error one called value? Which of those is correct and can we align them?

Comment (andrewoverton) [reply components/TextFields/src/MDCTextInputController.h:317]: I actually struggled with this a bit, it's a little tricky... `-setErrorText:errorAccessibilityValue:` sets the `accessibilityValue` on the `textInput` and the `accessibilityLabel` on the `leadingUnderlineLabel`. 

I wasn't sure if the same logic should underpin this method or not. I opted to try to do less than that method was doing and just set the `accessibilityLabel` on the `leadingUnderlineLabel`, and have the name reflect that.

FWIW, the MDCTextField computes _its_ `accessibilityValue` by [looking at](https://github.com/material-components/material-components-ios/blob/develop/components/TextFields/src/MDCTextField.m#L790) the `leadingUnderlineLabel` `accessibilityLabel`.

File: components/TextFields/src/MDCTextInputControllerBase.m [hunk 2]
```diff
@@ -987,10 +995,10 @@ + (void)setFloatingPlaceholderScaleDefault:(CGFloat)floatingPlaceholderScaleDefa
 
 - (void)setHelperText:(NSString *)helperText {
   if (self.isDisplayingErrorText) {
-    self.previousLeadingText = helperText;
+    self.previousLeadingText = [helperText copy];
```
Comment (jverkoey) [inline components/TextFields/src/MDCTextInputControllerBase.m]: Does this `copy` change need to land in this PR? Might be good to do in a separate PR.

Comment (andrewoverton) [reply components/TextFields/src/MDCTextInputControllerBase.m]: I can go ahead and take it out 👍

File: components/TextFields/src/MDCTextInputController.h (hunk 1 above)
Comment (ianegordon) [reply components/TextFields/src/MDCTextInputController.h:317]: While I strive for consistency in our APIs, in this particular case we are actually doing two different things.  In the error case we are intentionally setting the accessibilityValue.  In the helper case we are intentionally setting the accessibilityLabel.

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
LGTM.  Please verify the transitions between normal state (display / announce helperLabel) <> error state (display / announce errorValue) work as go back and forth between the two before merge.

**Comment by andrewoverton** [comment]
@ianegordon, just tested with the changes in 426069e and it seems to work as expected. Assuming that commit doesn't break any tests I'll merge once it passes 👍

**Pull request merged**